controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -432,4 +449,25 @@ void dynamic_channel_selection_task::fsm_move_state(eState new_state)
                     << s_ar_states[int(new_state)];
     m_fsm_state = new_state;
 }
-bool dynamic_channel_selection_task::fsm_in_state(eState state) { return m_fsm_state == state; }
\ No newline at end of file
+bool dynamic_channel_selection_task::fsm_in_state(eState state) { return m_fsm_state == state; }
+
+bool dynamic_channel_selection_task::start_scan()
+{
+    if (m_scanning_task_id != -1) {
+        return false;
+    }
+    // set scanning task id to current task id
+    m_scanning_task_id = id;
+    return true;
+}
+
+bool dynamic_channel_selection_task::finish_scan()
+{
+    if (m_scanning_task_id != id) {
+        TASK_LOG(ERROR) << "finish scan is called for a task that isn't the scanning task";
+        return false;
+    }
+    // scan by current task is aborted
+    m_scanning_task_id = -1;
```

Comment:
I'm pretty sure that a task can be erased without going through finish_scan().
I'm wondering if this is the only place for clearing this flag.

maybe we should use non default destructor and clear the flag also there
` ~dynamic_channel_selection_task() { m_scanning_task_id = -1 }`

Reply:
That's a good point, but instead of setting `m_scanning_task_id = -1;`, you should just call `finish_scan();` from the destructor (as it also make checks that the current task if the owner of this flag).

Reply:
added a fixup commit that calls `finish_scan();` as part of the DTOR - please re-review